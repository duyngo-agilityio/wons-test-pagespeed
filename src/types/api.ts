export enum Method {
  Get = 'GET',
  Post = 'POST',
  Put = 'PUT',
  Patch = 'PATCH',
  Delete = 'DELETE',
}
