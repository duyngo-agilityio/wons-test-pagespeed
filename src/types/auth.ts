// Models
import { User } from '@/models';

export interface SignInFormData {
  identifier: string;
  password: string;
}

export interface AuthResponse {
  jwt: string;
  user: User | null;
  error: string;
}

export type TSignUpPayload = {
  fullName: string;
  username: string;
  email: string;
  password: string;
};

export type TSignUpResponse = { jwt: string; user: Omit<User, 'password'> };
