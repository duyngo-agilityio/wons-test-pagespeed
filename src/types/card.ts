export enum Level {
  Low = 'low',
  Medium = 'medium',
  High = 'high',
}
