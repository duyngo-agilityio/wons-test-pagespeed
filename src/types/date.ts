import { DateValue } from '@internationalized/date';

export interface DateRangeState {
  start: DateValue;
  end: DateValue;
}
