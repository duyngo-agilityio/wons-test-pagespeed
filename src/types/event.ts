// Models
import { ICalendarTask, IEvent, TUser } from '@/models';

// Types
import { StrapiModel } from './strapi';

export type TEventResponse = Omit<IEvent, 'users_permissions_users'> &
  ICalendarTask & {
    users_permissions_users: { data: StrapiModel<TUser>[] };
  };
