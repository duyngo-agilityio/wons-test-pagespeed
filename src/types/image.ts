export interface UploadImageResponse {
  downloadURL?: string;
  error?: string;
}
