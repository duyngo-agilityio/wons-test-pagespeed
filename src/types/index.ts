export * from './auth';
export * from './date';
export * from './params';
export * from './select';
export * from './strapi';
export * from './table';
export * from './toast';
export * from './invoice';
export * from './customer';
