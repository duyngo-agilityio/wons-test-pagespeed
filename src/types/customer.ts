// Models
import { ICustomer } from '@/models';

// Types
import { StrapiModel, StrapiResponse } from '@/types';

export type TCustomerDataResponse = StrapiModel<ICustomer>;

export type TCustomerListResponse = StrapiResponse<TCustomerDataResponse[]>;
