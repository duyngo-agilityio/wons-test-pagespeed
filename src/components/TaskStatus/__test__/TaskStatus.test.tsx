import { render } from '@testing-library/react';
import TaskStatusComponent from '../index';

// Types
import { TLabelStatus } from '@/types';

describe('TaskStatusComponent Snapshot', () => {
  it('matches snapshot with "todo" status', () => {
    const { container } = render(
      <TaskStatusComponent status={'todo' as TLabelStatus} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with default styles for invalid status', () => {
    const { container } = render(
      <TaskStatusComponent status={'invalidStatus' as TLabelStatus} />,
    );
    expect(container).toMatchSnapshot();
  });
});
