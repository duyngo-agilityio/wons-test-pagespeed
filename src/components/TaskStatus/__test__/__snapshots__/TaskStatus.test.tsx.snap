// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskStatusComponent Snapshot matches snapshot with "todo" status 1`] = `
<div>
  <div
    class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap h-7 text-small text-default-foreground w-[100px] px-2 rounded-[20px] bg-blue-500/10 dark:bg-purple-600/10"
  >
    <span
      class="flex-1 font-normal px-2 text-center text-md capitalize text-blue-500 dark:text-purple-600"
    >
      todo
    </span>
  </div>
</div>
`;

exports[`TaskStatusComponent Snapshot matches snapshot with default styles for invalid status 1`] = `
<div>
  <div
    class="relative max-w-fit min-w-min inline-flex items-center justify-between box-border whitespace-nowrap h-7 text-small text-default-foreground w-[100px] px-2 rounded-[20px] bg-gray-300"
  >
    <span
      class="flex-1 font-normal px-2 text-center text-md capitalize text-gray-700"
    >
      invalidStatus
    </span>
  </div>
</div>
`;
